stock_archive_constraint/models/product_product.py

```diff
@@ -21,9 +21,10 @@ def _check_active_stock_archive_constraint_stock_quant(self):
         if res:
             raise ValidationError(
                 _(
-                    "It is not possible to archive product '%s' which has "
-                    "associated stock quantities." % res[0].product_id.display_name
+                    "It is not possible to archive product '%(display_name)s' which has "
+                    "associated stock quantities."
                 )
+                % {"display_name": res[0].product_id.display_name}
```

Comment:
You can use recordset directly as there is a limit=1.

`res.product_id.display_name`

Same for other lines